### ISSUE TITLE
networking.k8s.io/v1 Ingress is unavailable in v1.18.x

### ISSUE THREAD

**Issue by jeisonarenales** [kubecost/cost-analyzer-helm-chart#1216]
**Describe the bug**
It is not possible to install the chart(since version v1.88+) in K8s v1.18.x
This is because the chart uses APIs (networking.k8s.io/v1 for ingress) that is not available in v1.18.x

Specifically the Ingress https://kubernetes.io/docs/reference/using-api/deprecation-guide/#ingressclass-v122
Reference to your code: https://github.com/kubecost/cost-analyzer-helm-chart/blob/v1.88.0/cost-analyzer/templates/cost-analyzer-ingress-template.yaml#L7-L15

`Error: unable to build kubernetes objects from release manifest: unable to recognize "": no matches for kind "Ingress" in version "networking.k8s.io/v1"`

**To Reproduce**
1. Try to install the chart with version v1.88 or newer in a cluster running K8s v1.18.x

**Expected behavior**
Be able to install the chart.

**Screenshots**
Not relevant

**Collect logs (please complete the following information):**

 Not relevant

I would recommend using these validations instead: https://github.com/prometheus-community/helm-charts/blob/main/charts/alertmanager/templates/ingress.yaml#L9-L16

gz#1296

**Comment by AjayTripathy** [comment]
Hi @jeisonarenales I've seen this issue before and it's often the case that your kubectl CLI is running ahead of your kubernetes server version. Could you run `kubectl version` and let me know what the output is?

**Comment by jeisonarenales** [comment]
Hi @AjayTripathy, for sure here is my `kubectl version`
```
kubectl version
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.20", GitCommit:"1f3e19b7beb1cc0110255668c4238ed63dadb7ad", GitTreeState:"clean", BuildDate:"2021-06-16T12:58:51Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.20", GitCommit:"1f3e19b7beb1cc0110255668c4238ed63dadb7ad", GitTreeState:"clean", BuildDate:"2021-06-16T12:51:17Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}

kubectl explain ingress --api-version networking.k8s.io/v1
error: Couldn't find resource for "networking.k8s.io/v1, Kind=Ingress"
```

The `networking.k8s.io/v1` for `Ingress` is available since `v1.19.0+` https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.19.md#api-change-4

_Ingress and IngressClass resources have graduated to networking.k8s.io/v1. Ingress and IngressClass types in the extensions/v1beta1 and networking.k8s.io/v1beta1 API versions are deprecated and will no longer be served in 1.22+. Persisted objects can be accessed via the networking.k8s.io/v1 API. Notable changes in v1 Ingress objects (v1beta1 field names are unchanged):_

For this reason, your validation https://github.com/kubecost/cost-analyzer-helm-chart/blob/v1.88.0/cost-analyzer/templates/cost-analyzer-ingress-template.yaml#L7 is not working for `v1.18.x`, because `v1.18.x` has `networking.k8s.io/v1` but not for `Kind=Ingress`.

**Comment by AjayTripathy** [comment]
Ah, thank you so much for that detailed report! We'll get this fixed in our next release.

**Comment by AjayTripathy** [comment]
Assigned to support rotation engineer.

**Comment by jeisonarenales** [comment]
Closing this issue because it was solved in version [v1.90.0](https://github.com/kubecost/cost-analyzer-helm-chart/releases/tag/v1.90.0)
